File: 926.flip-string-to-monotone-increasing/solution.kt
class Solution {
    fun minFlipsMonoIncr(s: String): Int {
        var ones = 0
        var flips = 0
        for (ch in s) {
            if (ch == '0') {
                flips++
            } else {
                ones++
            }
            flips = minOf(flips, ones)
        }
        return flips
    }
}

File: 926.flip-string-to-monotone-increasing/solution.go
func minFlipsMonoIncr(s string) int {
    flips, ones := 0, 0
    for _, r := range s {
        if r == '0' {
            flips++
        } else {
            ones++
        }
        if ones < flips {
            flips = ones
        }
    }
    return flips
}